src/Payments/PazeButton.res

```diff
@@ -5,70 +5,65 @@ let make = (~token: SessionsType.token) => {
   let {iframeId, publishableKey, clientSecret} = Recoil.useRecoilValueFromAtom(keys)
   let {themeObj} = Recoil.useRecoilValueFromAtom(configAtom)
   let options = Recoil.useRecoilValueFromAtom(optionAtom)
-  let (showLoader, setShowLoader) = React.useState(() => false)
-  let setIsShowOrPayUsing = Recoil.useSetRecoilState(RecoilAtoms.isShowOrPayUsing)
+  let setIsShowOrPayUsing = Recoil.useSetRecoilState(isShowOrPayUsing)
   let loggerState = Recoil.useRecoilValueFromAtom(loggerAtom)
-  let intent = PaymentHelpers.usePaymentIntent(Some(loggerState), Paze)
   let isManualRetryEnabled = Recoil.useRecoilValueFromAtom(isManualRetryEnabled)
+  let intent = PaymentHelpers.usePaymentIntent(Some(loggerState), Paze)
   let paymentIntentID = clientSecret->Option.getOr("")->getPaymentId
-
-  React.useEffect0(() => {
-    setIsShowOrPayUsing(_ => true)
-    None
-  })
-
-  let onClick = _ => {
+  let (showLoader, setShowLoader) = React.useState(() => false)
+  let onClick = React.useCallback(_ => {
```

Comment:
usecallback can be little tricky if you forgot to add a dependency.

Should we try using react compiler to automate this process instead of manually adding dependencies and keeping them updated.